eixampledb-core/src/main/java/com/eixampledb/core/api/KeyTree.java

```diff
@@ -0,0 +1,57 @@
+package com.eixampledb.core.api;
+
+import java.util.TreeSet;
+import java.util.NavigableSet;
+import java.util.Collection;
+
+public class KeyTree {
```

Comment:
Don't think this class is strictly needed given that just forwards most of the calls to a TreeSet.